packages/loader/container-loader/src/createAndLoadContainerUtils.ts

```diff
@@ -173,3 +193,91 @@ export async function loadExistingContainer(
 		loadExistingContainerProps.pendingLocalState,
 	);
 }
+
+/**
+ * Loads a summarizer container with the required headers, triggers an on-demand summary, and then closes it.
+ * Returns success/failure and an optional error for host-side handling.
+ *
+ * @beta
+ */
+export async function loadSummarizerContainerAndMakeSummary(
+	loadExistingContainerProps: ILoadExistingContainerProps,
+): Promise<{ success: true } | { success: false; error: Error }> {
+	const loader = new Loader(loadExistingContainerProps);
+	const baseHeaders = loadExistingContainerProps.request.headers;
+	const request = {
+		...loadExistingContainerProps.request,
+		headers: {
+			...baseHeaders,
+			[LoaderHeader.cache]: false,
+			[LoaderHeader.clientDetails]: {
+				capabilities: { interactive: false },
+				type: "summarizer",
+			},
+			[DriverHeader.summarizingClient]: true,
+			[LoaderHeader.reconnect]: false,
+		},
+	};
+
+	const container = await loader.resolve(request);
+
+	loadExistingContainerProps.logger?.send({
+		category: "generic",
```

Comment:
Also, see how we use mixinMonitoringContext and createChildMonitoringContext in loader.ts to fill feature gate values from configProvider and also to provide namespace etc and then use that logger and configprovider.